### DISCUSSION TITLE
kafka cluster  can not use storage class on azure aks

### DISCUSSION THREAD

**Discussion by chanvoyage** [strimzi/strimzi-kafka-operator#5203] [Q&A]
hello , everyone

i have some problem  when deploy kafka on aks ,  kafka  broker status pending   it seems can not use storage class: managed-premium
=========================. yaml==============
apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: kafka-cluster
  namespace: aia-kafka
spec:
  kafka:
    version: 2.8.0
    replicas: 3
    listeners:
      - name: plain
        port: 9092
        type: internal
        tls: false
      - name: tls
        port: 9093
        type: internal
        tls: true
      - name: external
        port: 9094
        type: ingress
        tls: false
    config:
      offsets.topic.replication.factor: 3
      transaction.state.log.replication.factor: 3
      transaction.state.log.min.isr: 2
      log.message.format.version: "2.8"
      inter.broker.protocol.version: "2.8"
    storage:
      type: jbod
      class: managed-premium
      volumes:
      - id: 0
        type: persistent-claim
        size: 20Gi
        deleteClaim: false
      - id: 1
        type: persistent-claim
        size: 20Gi
        deleteClaim: false
      - id: 2
        type: persistent-claim
        size: 20Gi
        deleteClaim: false
  zookeeper:
    replicas: 3
    storage:
      type: persistent-claim
      size: 50Gi
      class: managed-premium
      deleteClaim: false
  entityOperator:
    topicOperator: {}
    userOperator: {}

![image](https://user-images.githubusercontent.com/39899497/123509038-7705a400-d69d-11eb-8e14-36e07101b46e.png)

![image](https://user-images.githubusercontent.com/39899497/123509047-81c03900-d69d-11eb-8ea5-2496aa69677b.png)

![image](https://user-images.githubusercontent.com/39899497/123508956-d57e5280-d69c-11eb-9e37-a9dc7bc7d0e8.png)

**Comment by scholzj** [comment]
You code example is hard to read. Please format it as a code. But I suspect you have the `class` in a wrong place: it should be specified for each volume, not at the upper level. You can see that also from the JBOD storage docs: https://strimzi.io/docs/operators/latest/full/using.html#type-JbodStorage-reference
